Builds fail on M1(ARM) Macs

When building `skia-safe` on my arm Mac, I get the error
```
error: unknown target triple 'unknown-apple-macosx11.0.0', please use -triple or -arch
ninja: build stopped: subcommand failed.
```
even though the clang invocation contains `-target arm64-apple-macos11` and `--target=aarch64-apple-darwin`

A similar error is mentioned [here](https://zulip-archive.rust-lang.org/242906tcompilerarm/44930macOSInitialCIconfiguration.html).

[full output](https://github.com/rust-skia/rust-skia/files/5936181/cargo.txt)

i think i am having the same issue:

```
% git clone https://github.com/rust-skia/rust-skia
% cd rust-skia ; cargo b
[...]
  clang++ -MD -MF obj/src/codec/jpeg_decode.SkJpegCodec.o.d -DNDEBUG -DSK_SUPPORT_GPU=0 -DSK_ASSUME_GL=1 -DSK_ENABLE_API_AVAILABLE -DSK_GAMMA_APPLY_TO_A8 -DSKIA_IMPLEMENTATION=1 -DSK_CODEC_DECODES_JPEG -I../../../../../../skia-bindings/skia -Wno-attributes -fstrict-aliasing -fPIC -fvisibility=hidden -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX11.1.sdk -target arm64-apple-macos11 -O3 -Wno-sign-conversion -Wno-unused-parameter --target=aarch64-apple-darwin -O0 -isystem /private/tmp/ack/rust-skia/skia-bindings/skia/third_party/libjpeg-turbo -isystem /private/tmp/ack/rust-skia/skia-bindings/skia/third_party/externals/libjpeg-turbo -std=c++17 -fvisibility-inlines-hidden -fno-exceptions -fno-rtti -c ../../../../../../skia-bindings/skia/src/codec/SkJpegCodec.cpp -o obj/src/codec/jpeg_decode.SkJpegCodec.o
  error: unknown target triple 'unknown-apple-macosx11.0.0', please use -triple or -arch
  ninja: build stopped: subcommand failed.
[...]
```

I noticed that clang++ gets passed `-target arm64-apple-macos11` and `--target=aarch64-apple-darwin` which seems to be the underlying issue.

Digging deeper I traced this arg to this part in `skia-bindings/build_support/skia.rs`:

```
@@ -281,7 +281,7 @@ impl FinalBuildConfiguration {
             let target_str: &str = &format!("--target={}", target.to_string());
             let sysroot_arg;
             let opt_level_arg;
-            let mut cflags: Vec<&str> = vec![&target_str];
+            let mut cflags: Vec<&str> = vec![];
```

With this change it builds for me:

```
% cargo clean
% cargo b
[...]
warning: 4 warnings generated.
   Compiling skia-safe v0.38.3 (/private/tmp/ack/rust-skia/skia-safe)
   Compiling skia-org v1.0.0 (/private/tmp/ack/rust-skia/skia-org)
    Finished dev [unoptimized + debuginfo] target(s) in 1m 27s
```

Now I am sure this approach breaks something somewhere else ;), but I am new to rust and this build process goes way over my head. I am happy to try out any PR that someone might come up with based on this finding.

--- a/skia-bindings/build_support/skia.rs
+++ b/skia-bindings/build_support/skia.rs
@@ -279,10 +279,11 @@ impl FinalBuildConfiguration {
             // target specific gn args.
             let target = cargo::target();
             let target_str: &str = &format!("--target={}", target.to_string());
+            let mut set_target = true;
             let sysroot_arg;
             let opt_level_arg;
-            let mut cflags: Vec<&str> = vec![&target_str];
-            let asmflags: Vec<&str> = vec![&target_str];
+            let mut cflags: Vec<&str> = vec![];
+            let mut asmflags: Vec<&str> = vec![];
 
             if let Some(sysroot) = cargo::env_var("SDKROOT") {
                 sysroot_arg = format!("--sysroot={}", sysroot);
@@ -346,7 +347,13 @@ impl FinalBuildConfiguration {
                 }
                 (arch, _, os, _) => {
                     let skia_target_os = match os {
-                        "darwin" => "mac",
+                        "darwin" => {
+                            // Skia will take care to set a specific `-target` for the current macOS
+                            // version. So we don't push another target `--target` that may
+                            // conflict.
+                            set_target = false;
+                            "mac"
+                        }
                         "windows" => "win",
                         _ => os,
                     };
@@ -360,6 +367,11 @@ impl FinalBuildConfiguration {
                 args.push(("skia_use_system_expat", yes_if(use_system_libraries)));
             } else {
                 args.push(("skia_use_expat", no()));
+            }
+
+            if set_target {
+                cflags.push(target_str);
+                asmflags.push(target_str);
             }
 
             if !cflags.is_empty() {

